src/main/java/org/folio/service/claiming/ClaimingService.java

```diff
@@ -0,0 +1,246 @@
+                                                                     RequestContext requestContext) {
+    var updatePiecesAndJobFutures = new ArrayList<Future<List<String>>>();
+    pieceIdsByVendorId.forEach((vendorId, pieceIds) -> config.stream()
+      .filter(entry -> isExportTypeClaimsAndCorrectVendorId(vendorId, entry) && Objects.nonNull(entry.getValue()))
+      .forEach(entry -> {
+        logger.info("createJobsByVendor:: Preparing job integration detail for vendor, vendor id: {}, pieces: {}, job key: {}",
+          vendorId, pieceIds.size(), entry.getKey());
+        updatePiecesAndJobFutures.add(updatePiecesAndCreateJob(pieceIds, entry, requestContext));
+      }));
+    return updatePiecesAndJobFutures;
+  }
+
+  private static ClaimingResults createEmptyClaimingResults(String message) {
+    return new ClaimingResults().withClaimingPieceResults(List.of(new ClaimingPieceResult().withError(new Error().withMessage(message))));
+  }
+
+  private List<ClaimingPieceResult> createSuccessClaimingResults(List<List<String>> updatedPieceLists) {
+    return updatedPieceLists.stream().flatMap(Collection::stream).distinct()
+        .map(pieceId -> new ClaimingPieceResult().withPieceId(pieceId).withStatus(SUCCESS))
+        .toList();
+  }
+
+  private List<ClaimingPieceResult> createErrorClaimingResults(Map<String, List<String>> pieceIdsByVendorId, String message) {
+    return pieceIdsByVendorId.values().stream()
+        .flatMap(Collection::stream)
+        .map(pieceId -> new ClaimingPieceResult().withPieceId(pieceId).withStatus(FAILURE).withError(new Error().withMessage(message)))
+        .toList();
+  }
+
+  private static boolean isExportTypeClaimsAndCorrectVendorId(String vendorId, Map.Entry<String, Object> entry) {
+    return entry.getKey().startsWith(String.format("%s_%s", EXPORT_TYPE_CLAIMS, vendorId));
+  }
+
+  private Future<List<String>> updatePiecesAndCreateJob(List<String> pieceIds, Map.Entry<String, Object> entry,
+                                                        RequestContext requestContext) {
+    logger.info("updatePiecesAndCreateJob:: Updating pieces and creating a job, job key: {}, count: {}", entry.getKey(), pieceIds.size());
+    return pieceUpdateFlowManager.updatePiecesStatuses(pieceIds, PieceBatchStatusCollection.ReceivingStatus.CLAIM_SENT, requestContext)
+        .compose(v -> createJob(entry.getKey(), entry.getValue(), pieceIds, requestContext).map(pieceIds));
+  }
+
+  private Future<Void> createJob(String configKey, Object configValue, List<String> pieceIds, RequestContext requestContext) {
+    var integrationDetail = new JsonObject(String.valueOf(configValue));
+    integrationDetail.getJsonObject(EXPORT_TYPE_SPECIFIC_PARAMETERS)
+      .getJsonObject(VENDOR_EDI_ORDERS_EXPORT_CONFIG)
+      .put(CLAIM_PIECE_IDS, pieceIds);
+    return restClient.post(resourcesPath(DATA_EXPORT_SPRING_CREATE_JOB), integrationDetail, Object.class, requestContext)
+      .map(response -> {
```

Comment:
per our latest finding compose is more appropriate here to avoid random async issues